dspace-api/src/test/java/org/dspace/app/util/ConfigurationIT.java

```diff
@@ -0,0 +1,227 @@
+    private static final String SINGLE_VALUE = "value";
+
+    private static final String ARRAY_PROPERTY = "test.array";
+    private static final String[] ARRAY_VALUE = { "one", "two" };
+
+    private static final String PLACEHOLDER_PROPERTY = "test.substituted";
+    private static final String PLACEHOLDER_VALUE = "insert ${test.single} here"; // Keep aligned with SINGLE_NAME
+    private static final String SUBSTITUTED_VALUE = "insert value here"; // Keep aligned with SINGLE_VALUE
+
+    private static final String MISSING_PROPERTY = "test.missing";
+
+    /** Capture standard output. */
+    @Rule
+    public final SystemOutRule systemOutRule = new SystemOutRule();
+
+    /** Capture standard error. */
+    @Rule
+    public final SystemErrRule systemErrRule = new SystemErrRule();
+
+    /** Capture System.exit() value. */
+    @Rule
+    public final ExpectedSystemExit expectedSystemExit = ExpectedSystemExit.none();
+
+    /**
+     * Create some expected properties before all tests.
+     */
+    @BeforeClass
+    public static void setupSuite() {
+        cfg = kernelImpl.getConfigurationService();
+
+        cfg.setProperty(SINGLE_PROPERTY, SINGLE_VALUE);
+        cfg.setProperty(ARRAY_PROPERTY, ARRAY_VALUE);
+        cfg.setProperty(PLACEHOLDER_PROPERTY, PLACEHOLDER_VALUE);
+        cfg.setProperty(MISSING_PROPERTY, null); // Ensure that this one is undefined
+    }
+
+    /**
+     * After all tests, remove the properties that were created at entry.
+     */
+    @AfterClass
+    public static void teardownSuite() {
+        if (null != cfg) {
+            cfg.setProperty(SINGLE_PROPERTY, null);
+            cfg.setProperty(ARRAY_PROPERTY, null);
+            cfg.setProperty(PLACEHOLDER_PROPERTY, null);
+        }
+    }
+
+    /**
+     * Test fetching all values of a single-valued property.
+     */
+    @Test
+    public void testMainAllSingle() {
+        String[] argv;
+        argv = new String[] {
+            "--property", SINGLE_PROPERTY
+        };
+        expectedSystemExit.expectSystemExitWithStatus(0);
+        expectedSystemExit.checkAssertionAfterwards(new Assertion() {
+            @Override public void checkAssertion() {
+                String[] output = systemOutRule.getLogWithNormalizedLineSeparator()
+                        .split("\n");
+                assertThat(output, arrayWithSize(1));
+            }
+        });
+        expectedSystemExit.checkAssertionAfterwards(new Assertion() {
+            @Override public void checkAssertion() {
+                String[] output = systemOutRule.getLogWithNormalizedLineSeparator()
+                        .split("\n");
+                assertThat(output[0], equalTo(SINGLE_VALUE));
+            }
+        });
+        systemOutRule.enableLog();
+        Configuration.main(argv);
+    }
+
+    /**
+     * Test fetching all values of an array property.
+     */
+    @Test
+    public void testMainAllArray() {
```

Comment:
Could we add a version of this Array test that uses the new `--first` flag?  It looks like it was accidently forgotten.

(As a sidenote, I noticed all the test methods in this IT are name "testMainAll*".  Since the "--all" flag was removed, we might want to remove the name "All" from the tests and just call them all "testMain*"

Reply:
I've added tests of array and single properties using `--first`.

I left the test names as-is, since the old ones all test the default all-values case.

Reply:
Tests for the first-value case, which was originally the only case, were deliberately left out of the original test set because I was being lazy and only writing tests for my new code.